server/bin/pbench-unpack-tarballs.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python3
+# -*- mode: python -*-
+
+import os
+import sys
+import glob
+import tempfile
+
+from logging import Logger
+from pathlib import Path
+
+from pbench.common.exceptions import BadConfig
+from pbench.common.logger import get_pbench_logger
+from pbench.server import PbenchServerConfig
+from pbench.server.filetree import FileTree
+from pbench.server.report import Report
+from pbench.server.database import init_db
+from pbench.server.utils import get_tarball_md5
+
+
+_NAME_ = "pbench-unpack-tarballs"
+
+# The link source for the operation of this script.
+_linksrc = "TO-UNPACK"
+
+
+def unpack_tarball(config: PbenchServerConfig, logger: Logger):
+
+    tarlist = glob.iglob(os.path.join(config.ARCHIVE, "*", _linksrc, "*.tar.xz"))
```

Comment:
I actually missed this until I looked again after seeing Webb's comment about legacy tests; but this code should follow the `pbench-unpack-tarballs.sh` logic of removing the `TO-UNPACK` symlinks and creating `TO-COPY-SOS` and `TO-INDEX` links to the tarball after it's done. Otherwise a server deployed with your change won't actually work ...

Reply:
I somehow thought that maybe filetree will take care of that. Will create the symlinks in those state directories as well.

Reply:
Ah. The `FileTree` class wasn't intended to worry about the "state" of datasets, just to help manipulate the physical artifacts (currently in the file tree). That is, if you call `unpack`, it should unpack the physical artifacts without worrying about the current "state" or assuming anything about the next "state".

Reply:
Oh.. ok I will add the symlinks to those states as well :)